File: backend/src/main/java/com/tribly/dto/posts/response/PostListResponse.java
package com.tribly.dto.posts.response;

import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated post list response")
public record PostListResponse(
    @Schema(description = "List of posts", required = true) List<PostDto> posts,
    @Schema(description = "Total number of posts", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/postListResponse.ts
import type { PostDto } from './postDto.ts'

/**
 * Paginated post list response
 */
export interface PostListResponse {
  /** List of posts */
  posts: PostDto[]
  /** Total number of posts */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
